packages/apps/job-launcher/server/src/modules/manifest/manifest.service.spec.ts

```diff
@@ -0,0 +1,851 @@
+      const mockData = { key: faker.lorem.word() };
+      const mockOracleAddresses: string[] = [];
+      const mockManifestData = {
+        url: faker.internet.url(),
+        hash: faker.string.uuid(),
+      };
+
+      mockStorageService.uploadJsonLikeData.mockResolvedValueOnce(
+        mockManifestData,
+      );
+
+      const result = await manifestService.uploadManifest(
+        mockChainId,
+        mockData,
+        mockOracleAddresses,
+      );
+
+      expect(result).toEqual(
+        expect.objectContaining({
+          url: mockManifestData.url,
+          hash: mockManifestData.hash,
+        }),
+      );
+    });
+
+    it('should throw an error if upload fails', async () => {
+      const mockChainId = faker.number.int();
+      const mockData = { key: faker.lorem.word() };
+      const mockOracleAddresses: string[] = [];
+
+      mockStorageService.uploadJsonLikeData.mockRejectedValue(
+        new ControlledError('File not uploaded', HttpStatus.BAD_REQUEST),
+      );
+
+      await expect(
+        manifestService.uploadManifest(
+          mockChainId,
+          mockData,
+          mockOracleAddresses,
+        ),
+      ).rejects.toThrow(ControlledError);
+    });
+  });
+
+  describe('downloadManifest', () => {
+    it('should download and validate a manifest successfully', async () => {
+      const mockManifestUrl = faker.internet.url();
+      const mockRequestType = FortuneJobType.FORTUNE;
+      const mockManifest: FortuneManifestDto = {
+        submissionsRequired: faker.number.int({ min: 1, max: 100 }),
+        requesterTitle: faker.lorem.words(3),
+        requesterDescription: faker.lorem.sentence(),
+        fundAmount: faker.number.float({ min: 1, max: 1000 }),
+        requestType: FortuneJobType.FORTUNE,
+        qualifications: [faker.lorem.word(), faker.lorem.word()],
+      };
+      mockStorageService.downloadJsonLikeData.mockResolvedValueOnce(
+        mockManifest,
+      );
+      const result = await manifestService.downloadManifest(
+        mockManifestUrl,
+        mockRequestType,
+      );
+      expect(result).toEqual(mockManifest);
+    });
+
+    it('should throw an error if validation fails', async () => {
+      const mockManifestUrl = faker.internet.url();
+      const mockRequestType = CvatJobType.IMAGE_BOXES;
+      const mockManifest: FortuneManifestDto = {
+        submissionsRequired: faker.number.int({ min: 1, max: 100 }),
+        requesterTitle: faker.lorem.words(3),
+        requesterDescription: faker.lorem.sentence(),
+        fundAmount: faker.number.float({ min: 1, max: 1000 }),
+        requestType: FortuneJobType.FORTUNE,
+        qualifications: [faker.lorem.word(), faker.lorem.word()],
+      };
+      mockStorageService.downloadJsonLikeData.mockResolvedValueOnce(
+        mockManifest,
+      );
+      await expect(
+        manifestService.downloadManifest(mockManifestUrl, mockRequestType),
+      ).rejects.toThrow(ControlledError);
```

Comment:
Maybe check that it throws `ManifestValidationFailed` error and not anything else?